metaflow/parameters.py

```diff
@@ -293,10 +293,19 @@ def __init__(
             "params",
             "with",
             "max-num-splits",
+            "max_num_splits",
             "max-workers",
+            "max_workers",
             "tag",
             "run-id-file",
+            "run_id_file",
             "namespace",
+            "obj",
+            "decospecs",
+            "max-log-size",
+            "max_log_size",
+            "tags",
+            "user_namespace",
```

Comment:
should we also have `user-namespace`? it might be easier if `reserved_params` is a set that adds the dashed versions of all its member by itself so that it is easy to maintain longer term.

Reply:
Agreed we should standardize _ and -

Reply:
done. Lifted the list construction to a separate function